Add additional array examples

AlanFoster opened the pull request:
Just a PR to document the behavior of Ruby arrays.

AlanFoster committed:
Add additional array examples

AlanFoster commented on tests/array/__snapshots__/jsfmt.spec.js.snap:
```diff
@@ -8,3 +8,41 @@ numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+symbols
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+def symbols
+  ['    a', '    b', '    c', '    d']
```
It seems that the Ripper sexpressions from Ruby include this whitespace:

```ruby
[:program,
 [[:def,
   [:@ident, "strings", [1, 4]],
   [:params, nil, nil, nil, nil, nil, nil, nil],
   [:bodystmt,
    [[:array,
      [[:@tstring_content, "    a", [3, 0]],
       [:@tstring_content, "    b", [4, 0]],
       [:@tstring_content, "    c", [5, 0]],
       [:@tstring_content, "    d", [6, 0]]]]],
    nil,
    nil,
    nil]],
  [:vcall, [:@ident, "strings", [10, 0]]]]]
```

But if you run it in an interpreter the white space isn't present:

```
>   strings
=> ["a", "b", "c", "d"]
```

I'm not sure what to make of this, other than it's not quite the behavior I expected 🤔

iamsolankiamit replied:
The sexp is correct, if you check the tokens array, you will find one of these symbols.

```ruby
:on_qwords_beg, :on_qsymbols_beg, :on_words_beg, :on_symbols_beg
```

What needs to be done is to keep on removing tokens from the tokens array wherever the need arises. So inside visit_array func we can check what is the current token and based on that we can format the `:@tstring_content` value to strip the whitespace if the current_token is `:on_words_beg` and so on

AlanFoster replied:
Creating a separate issue to keep track of this behavior:

https://github.com/iamsolankiamit/prettier-ruby/issues/6

AlanFoster commented:
@iamsolankiamit Noticed that arrays crashed when they're empty, so I've added an example of that - and fixed the crash.

I think this should be good to merge now that there's a separate ticket for keeping track of indented `%w( ... )` syntax.

iamsolankiamit commented:
@AlanFoster Can you update the snapshots to expected output, so we can have a failing test. I'll get this done by the weekend, thanks!

AlanFoster commented:
@iamsolankiamit I don't have a _strong_ preference on output, but I've added some examples to the original ticket:

https://github.com/iamsolankiamit/prettier-ruby/issues/6#issuecomment-393518869

I won't add failing tests as part of _this_ pull request though, which means I can get this merged in, master can remain green, and I can keep throwing up PRs :)
